Development.md

```diff
@@ -0,0 +1,250 @@
+# Convox Rack Development Guide
+
+Rack is open source with the goal of everyone to understand the platform, provide suggestions, and contribute improvements.
+
+This guide documents how to:
+
+* Set up a sandbox AWS Rack to develop and test infrastructure template changes
+* Set up a laptop with a Docker VM to run, develop and test API changes locally
+* Run the Rack unit test suite on a laptop
+* Submit changes upstream
+* Run an AWS integration test suite locally or on a CI server
+* Release artifacts to enable `convox rack update`
+
+## Sponsorship and Mentoring
+
+Developing Rack will incur AWS costs. If this is an obstacle for you to contribute you can contact [support@convox.com](mailto:support@convox.com) to request sponsorship.
+
+Much of the technical design and implementation in Rack requires understanding of AWS, Docker, Golang, systems engineering and more. If you would like to better learn these systems to contribute, you can contact support@convox.com, join the [Public Slack](http://invite.convox.com/), or open up issues on [GitHub](http://github.com/convox/rack) to ask questions and/or request a mentor.
+
+## Sandbox AWS Rack Install
+
+Rack consumes numerous AWS and Docker APIs. The easiest way to develop Rack is with real AWS access keys interacting with real AWS resources like a Dynamo Table, ECS Cluster, and CloudFormation Stack.
+
+This is easy to bootstrap with the Rack project itself:
+
+```
+$ convox install --stack-name dev
+```
+
+It can also be bootstrapped with no `convox` tools via the AWS CLI:
+
+```
+$ aws cloudformation create-stack --stack-name dev --template-body file://$(pwd)/api/dist/kernel.json
+```
+
+You can also use any existing Rack with the caveat that running a laptop Rack against it could have side effects like terminating instances.
+
+## AWS Rack Ingress
+
+Parts of the API like `convox ps --stats` interact with the Docker daemon running on every AWS instance. To enable `convox ps --stats` to work from a laptop you will want to open up access to the instance Docker daemons:
```

Comment:
Hm, possibly but we'd need a script that makes every instance docker host available on a local port range.